projects/sunbird-video-player/src/lib/services/viewer.service.ts

```diff
@@ -245,12 +245,15 @@ export class ViewerService {
       'RATE_CHANGE', 'CLOSE_DOWNLOAD', 'DOWNLOAD', 'NAVIGATE_TO_PAGE',
       'NEXT', 'OPEN_MENU', 'PREVIOUS', 'CLOSE_MENU', 'DOWNLOAD_MENU', 'DOWNLOAD_POPUP_CLOSE', 'DOWNLOAD_POPUP_CANCEL',
       'SHARE', 'REPLAY', 'FORWARD', 'BACKWARD', 'FULLSCREEN', 'NEXT_CONTENT_PLAY', 'TRANSCRIPT_LANGUAGE_OFF',
-      'TRANSCRIPT_LANGUAGE_SELECTED'
+      'TRANSCRIPT_LANGUAGE_SELECTED', 'VIDEO_MARKER_SELECTED'
     ];
     if (interactItems.includes(type)) {
       this.videoPlayerService.interact(type.toLowerCase(), 'videostage', extraValues);
     }
+  }
 
+  raiseImpressionEvent(pageId: string, questionSetId?: string) {
```

Comment:
here instead of questionSetId as param send as telemetry object, make it generic not specific to question set only

Reply:
@HarishGangula  Actually, we are creating object in the `sunbird-video-player.service.ts` with these values.

Reply:
Here the intention is if we have to generate another impression event without any questionSetId it will be multiple places code change since questionSetId is optional param so make it a generic param which suits the impression event